verilog/analysis/checkers/macro_string_concatenation_rule.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/macro_string_concatenation_rule.h"
+
+#include <string>
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
+#include "common/analysis/citation.h"
+#include "common/text/token_info.h"
+#include "common/util/value_saver.h"
+#include "verilog/analysis/descriptions.h"
+#include "verilog/analysis/lint_rule_registry.h"
+#include "verilog/parser/verilog_lexer.h"
+#include "verilog/parser/verilog_token_classifications.h"
+
+namespace verilog {
+namespace analysis {
+
+using verible::GetStyleGuideCitation;
+using verible::LintRuleStatus;
+using verible::LintViolation;
+using verible::TokenInfo;
+
+// Register the lint rule
+VERILOG_REGISTER_LINT_RULE(MacroStringConcatenationRule);
+
+absl::string_view MacroStringConcatenationRule::Name() {
+  return "macro-string-concatenation";
+}
+const char MacroStringConcatenationRule::kTopic[] = "defines";
+const char MacroStringConcatenationRule::kMessage[] =
+    "Token concatenation (``) used inside plain string literal.";
+
+std::string MacroStringConcatenationRule::GetDescription(
+    DescriptionType description_type) {
+  return absl::StrCat(
+      "Concatenation will not be evaluated here. Use `\"...`\" instead. "
+      "See ", GetStyleGuideCitation(kTopic), ".");
+}
+
+void MacroStringConcatenationRule::HandleToken(const TokenInfo& token) {
+  const auto token_enum = static_cast<verilog_tokentype>(token.token_enum());
+  const absl::string_view text(token.text());
+
+  // Search only in `define tokens. Ignore state as `defines can be nested.
+  if (token_enum == PP_define_body) {
+    if (IsUnlexed(verilog_tokentype(token.token_enum()))) {
```

Comment:
since you already have a properly casted `token_enum`, doing this function-style cast `verilog_tokentype(token.token_enum())` looks superfluous.